tests/test_rff.py

```diff
@@ -0,0 +1,538 @@
+    disc = "risk_aversion"
+    eta_rho = [1, 2]
+    file = "damage_functions"
+    mask = "unmasked"
+
+    df_in_dir = d / sector / pulse_year / mask
+    df_out_dir = d / "out" / sector / pulse_year / mask
+
+    df_in_file = (
+        df_in_dir / f"{recipe}_{disc}_eta{eta_rho[0]}_rho{eta_rho[1]}_{file}.nc4"
+    )
+    outfile_path = (
+        df_out_dir / f"{recipe}_{disc}_eta{eta_rho[0]}_rho{eta_rho[1]}_{file}.nc4"
+    )
+    fractional_path = (
+        df_out_dir
+        / f"{recipe}_{disc}_eta{eta_rho[0]}_rho{eta_rho[1]}_fractional_{file}.nc4"
+    )
+
+    Path(df_in_dir).mkdir(parents=True, exist_ok=True)
+    Path(df_out_dir).mkdir(parents=True, exist_ok=True)
+    df_in = xr.Dataset(
+        {
+            "anomaly": (
+                ["discount_type", "ssp", "model", "year"],
+                np.ones((1, 2, 2, 3)),
+            ),
+        },
+        coords={
+            "discount_type": (["discount_type"], ["euler_ramsey"]),
+            "ssp": (["ssp"], ["SSP3", "SSP4"]),
+            "model": (["model"], ["IIASA GDP", "OECD Env-Growth"]),
+            "year": (["year"], [2021, 2022, 2099]),
+        },
+    )
+    df_in.to_netcdf(df_in_file)
+
+    factors = xr.Dataset(
+        {
+            "anomaly": (
+                ["discount_type", "ssp", "model", "year"],
+                np.ones((1, 2, 2, 3)),
+            ),
+        },
+        coords={
+            "discount_type": (["discount_type"], ["euler_ramsey"]),
+            "ssp": (["ssp"], ["SSP3", "SSP4"]),
+            "model": (["model"], ["IIASA GDP", "OECD Env-Growth"]),
+            "year": (["year"], [2100, 2101, 2102]),
+        },
+    )
+    df_in.to_netcdf(df_in_file)
+
+    out_expected = xr.Dataset(
+        {
+            "anomaly": (
+                ["discount_type", "year", "ssp", "model"],
+                np.ones((1, 6, 2, 2)),
+            ),
+        },
+        coords={
+            "discount_type": (["discount_type"], ["euler_ramsey"]),
+            "year": (["year"], [2021, 2022, 2099, 2100, 2101, 2102]),
+            "ssp": (["ssp"], ["SSP3", "SSP4"]),
+            "model": (["model"], ["IIASA GDP", "OECD Env-Growth"]),
+        },
+    )
+
+    out_expected_fractional = xr.Dataset(
+        {
+            "anomaly": (["discount_type", "year"], np.ones((1, 3))),
+        },
+        coords={
+            "discount_type": (["discount_type"], ["euler_ramsey"]),
+            "year": (["year"], [2021, 2022, 2099]),
+        },
+    )
+
+    weight_df(
+        sector,
+        eta_rho,
+        recipe,
+        disc,
+        file,
+        d,
+        d / "out",
+        1,
+        2,
+        0.5,
+        factors,
+        pulse_year,
+        True,
+        mask="unmasked",
+    )
+
+    xr.testing.assert_allclose(out_expected, xr.open_dataset(outfile_path))
+    xr.testing.assert_allclose(
+        out_expected_fractional, xr.open_dataset(fractional_path)
+    )
+
+
+def test_rff_damage_functions(tmp_path, save_ssprff_econ):
```

Comment:
Do we test this with `USA=True` anywhere? Should we?

Reply:
This is some pretty basic selection functionality that I didn't think we needed to test since it is effectively tested for in so many other places.